packages/stream_chat_flutter/lib/src/message_modal/message_modal.dart

```diff
@@ -168,22 +168,14 @@ class StreamMessageModal extends StatelessWidget {
 extension ColumnAlignmentExtension on AlignmentGeometry {
   /// Converts an [AlignmentGeometry] to the most appropriate
   /// [CrossAxisAlignment] value.
-  CrossAxisAlignment toCrossAxisAlignment() {
-    return switch (this) {
-      // Center alignments
-      Alignment.topCenter => CrossAxisAlignment.start,
-      Alignment.center => CrossAxisAlignment.center,
-      Alignment.bottomCenter => CrossAxisAlignment.center,
-      // Left alignments
-      Alignment.topLeft => CrossAxisAlignment.start,
-      Alignment.centerLeft => CrossAxisAlignment.start,
-      Alignment.bottomLeft => CrossAxisAlignment.end,
-      // Right alignments
-      Alignment.topRight => CrossAxisAlignment.start,
-      Alignment.centerRight => CrossAxisAlignment.end,
-      Alignment.bottomRight => CrossAxisAlignment.end,
-      // Fallback
-      _ => CrossAxisAlignment.center,
+  CrossAxisAlignment toCrossAxisAlignment(BuildContext context) {
+    final resolved = resolve(Directionality.of(context));
+
+    return switch (resolved.x) {
+      0.0 => CrossAxisAlignment.center,
+      < 0 => CrossAxisAlignment.start,
+      > 0 => CrossAxisAlignment.end,
```

Comment:
I think now you base this on `x` you should just use left right, because otherwise if you set `Alignment.end` on RTL devices you will get x<0 and you change it into start.

```suggestion
      0.0 => CrossAxisAlignment.center,
      < 0 => CrossAxisAlignment.left,
      > 0 => CrossAxisAlignment.right,
```

Reply:
Yes, but that wouldn't be a problem because we resolve it first.

```dart
  CrossAxisAlignment toCrossAxisAlignment(BuildContext context) {
    final resolved = resolve(Directionality.of(context));

    return switch (resolved.x) {
      0.0 => CrossAxisAlignment.center,
      < 0 => CrossAxisAlignment.start,
      > 0 => CrossAxisAlignment.end,
      _ => CrossAxisAlignment.center, // fallback (in case of NaN etc)
    };
  }
```